server/src/HttpServer.hs

```diff
@@ -2,38 +2,39 @@
 
 module HttpServer (new) where
 
+import Control.Monad
 import Control.Monad.IO.Class
 import Network.HTTP.Types
 import Network.Wai (Application)
-import Web.Scotty (delete, get, json, jsonData, put, regex, request, scottyApp, status, ActionM)
+import Web.Scotty (delete, get, json, jsonData, put, regex, middleware, request, scottyApp, status, ActionM)
 import qualified Network.Wai as Wai
 
-import Core (Core, EnqueueResult (..))
-import Guardian(protected)
+import JwtMiddleware (jwtMiddleware)
+import Core (Core (..), EnqueueResult (..))
+import Config (Config (..))
 import qualified Core
 
-
 new :: Core -> IO Application
 new core =
   scottyApp $ do
+    when (configEnableJwtAuth $ coreConfig core) $
+      middleware $ jwtMiddleware $ configJwtSecret $ coreConfig core
```

Comment:
So many dollars :speak_no_evil: 
It doesn't really matter but I guess most of us would do:
``` haskell
(middleware . jwtMiddleware . configJwtSecret . coreConfig) core
```